src/sulogin-shell/sulogin-shell.c

```diff
@@ -89,15 +90,28 @@ static void print_mode(const char* mode) {
 }
 
 int main(int argc, char *argv[]) {
-        static const char* const sulogin_cmdline[] = {SULOGIN, NULL};
+        /* Note: sulogin_cmdline double-NULL is intentional!
+         * (The first null might be overwritten with --force argument)
+         */
+        static const char* sulogin_cmdline[] = {SULOGIN, NULL, NULL};
         _cleanup_(sd_bus_flush_close_unrefp) sd_bus *bus = NULL;
         int r;
+        const char *mode;
 
         log_set_target(LOG_TARGET_AUTO);
         log_parse_environment();
         log_open();
 
-        print_mode(argc > 1 ? argv[1] : "");
+        mode = argc > 1 ? argv[1] : "";
+        print_mode(mode);
+
+        if (streq(mode, "rescue") && getenv_bool("ENABLE_FORCED_SULOGIN") > 0) {
+                assert (sulogin_cmdline[2] == NULL);
```

Comment:
coding style: no space between `assert` and `(`...

that said, i'd drop this assert entirely, the definition of the array is just a few lines above after all.. Also, just checking `[2]` is kinda incomplete...

Reply:
The thought was just to have a safety-guard before overwriting the argument because I'm willing to bet someone at some point will be "cleaning up" the double-NULLs. Maybe I went way overboard both with the big comment banner and the assert together though. Using the style you suggested above mentioning /* --force */ right after the null should be clear enough and at the same time discrete. I'd still think it's a useful assert though, but if you prefer I'll drop it because I don't really care that much.

Reply:
but you are checking `[2]` here, even if you override `[1]`, or I am not sure I follow?

either way, I think we should drop this line